community/kernel/src/test/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStoreTest.java

```diff
@@ -92,17 +97,23 @@ public String toString()
 
     @Test
     @Resources.Life( STARTED )
+    @SuppressWarnings( "unchecked" )
     public void retryLookupOnConcurrentStoreStateChange() throws IOException
     {
-        Store testStore = resourceManager.managed( createTestStore( TimeUnit.DAYS.toMillis( 2 ) ) );
-        ConcurrentMapState<String> newState = new ConcurrentMapState<>( testStore.state, mock( File.class ), EmptyVersionContextSupplier.EMPTY );
-        testStore.put( "test", "value" );
+        Store store = resourceManager.managed( new Store() );
+
+        ProgressiveState<String> workingState = stateWithLookup( () -> true );
+        ProgressiveState<String> staleState = stateWithLookup( () -> {
+            setState( store, workingState );
+            throw new FileIsNotMappedException( new File( "/files/was/rotated/concurrently/during/lookup" ) ); } );
```

Comment:
It's slightly better now, because if the page cache exception is changed again it is much easier to see where it is catched compared to `IllegalStateException`. I do want to add a lookup/rotate stress test as well, once SOAKs are stable again.